libs/coin-bitcoin/src/wallet-btc/crypto/base.ts

```diff
@@ -29,8 +29,8 @@ export function fallbackValidateAddress(address: string): boolean {
 
 class Base implements ICrypto {
   network: any;
-  protected static bip32Cache = {}; // xpub + account + index to publicKey
-  public static addressCache = {}; // derivationMode + xpub + account + index to address
+  protected static bip32Cache: Record<string, Promise<BIP32>> = {}; // xpub + account + index to publicKey
```

Comment:
We can discuss it on slack, where comes the need to change BIP32 => Promise<BIP32> ? :thinking: is it because of the sctricter typing environment you were talking about ?

Reply:
The need is related to the consistency of the code (ex: line 71).
Sometimes we stored Promises, sometimes directly BIP32.
We can switch to BIP32, I don't have a strong opinion on that.

Reply:
`accountLevelP = Base.bip32Cache[keyAccount]; // it's stored as promise`

Is it what you are talking about ?